From 6ae5ecfde2f2ff4d3a0606d21a6cc2b5384413a4 Mon Sep 17 00:00:00 2001
From: Antonios Christidis <a-christidis@ti.com>
Date: Wed, 19 Feb 2025 13:45:16 -0600
Subject: [PATCH 86/87] PENDING: arm64: dts: ti: k3-j721e-main: add gpu node

Add the IMG 8XE-GE8430 GPU device node. This cannot be upstreamed as
this device currently uses an out-of-tree kernel module.

Add clock and power domains to the GPU node for J721E. This brings the
dust power domain control back under the driver and allows users to
override the default clock rate for the GPU's core clock.

Signed-off-by: Antonios Christidis <a-christidis@ti.com>
---
 arch/arm64/boot/dts/ti/k3-j721e-main.dtsi | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
index 67dd56477076..eed685ab136a 100644
--- a/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721e-main.dtsi
@@ -2615,6 +2615,19 @@ icssg1_mdio: mdio@32400 {
 		};
 	};
 
+	gpu: gpu@4e20000000 {
+		 compatible = "ti,j721e-pvr", "img,pvr-ge8430";
+		 reg = <0x4e 0x20000000 0x00 0x80000>;
+		 interrupts = <GIC_SPI 24 IRQ_TYPE_LEVEL_HIGH>;
+		 power-domains = <&k3_pds 125 TI_SCI_PD_EXCLUSIVE>,
+				 <&k3_pds 126 TI_SCI_PD_EXCLUSIVE>;
+		 power-domain-names = "firmware", "dust";
+		 clocks = <&k3_clks 125 0>;
+		 assigned-clocks = <&k3_clks 125 0>;
+		 assigned-clock-rates = <750000000>;
+		 clock-names = "core";
+	};
+
 	main_mcan0: can@2701000 {
 		compatible = "bosch,m_can";
 		reg = <0x00 0x02701000 0x00 0x200>,
-- 
2.39.5 (Apple Git-154)

